### ISSUE TITLE
Hidden elements

### ISSUE THREAD

**Issue by HugoGiraudel** [HugoGiraudel/a11y.js#4]
I was thinking that we could bubble hidden issues on the first displayed parent. Let's consider a rule like so:

```css
[hidden]:not([aria-hidden])
```

Because the `hidden` attribute hides the element, we cannot hover it to display the message. An idea could be to bubble up in the tree to apply the data-a11y attributes to its first ancestor that is displayed, as well as an extra attribute to explain where does this message come from.

```js
Object.keys(this.conf.data).forEach(function (selector) {
      var $selector = $(selector);
      // For each selector, grab all matching nodes
      $selector.toArray().forEach(function (element) {
        var $element = $(element);
        var bubble = false;

        if ($element.is(':hidden')) {
          $element = $element.closest(':not(:hidden)'); // Not tested, but should work
          bubble = true;
        }
        
        // And add them some attributes
        $element
          .attr('data-a11y-type', this.conf.data[selector].type)
          .attr('data-a11y-message', this.conf.data[selector].message);
       
        // If message has bubbled up the parent, add an extra attribute
        if (bubble) {
          $element.attr('data-a11y-from', selector);
        }
      }, this);
    }, this);
  };
```
